src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -1002,14 +1004,157 @@ namespace
             asr::SunLightFactory().create(
                 light_name.c_str(),
                 asr::ParamArray()
-                  .insert("radiance_multiplier", intensity * asf::Pi<float>())
-                  .insert("turbidity", 1.0)
-                  .insert("environment_edf", sky_name)
-                  .insert("size_multiplier", size_mult)));
+                .insert("radiance_multiplier", intensity * asf::Pi<float>())
+                .insert("turbidity", 1.0)
+                .insert("environment_edf", sky_name)
+                .insert("size_multiplier", size_mult)));
         light->set_transform(transform);
         assembly.lights().insert(light);
     }
 
+    void add_analytical_light(
+        asr::Assembly&              assembly,
+        const std::string&          light_name,
+        const Matrix3&              light_transform,
+        const std::string&          color_name,
+        const float                 intensity,
+        asr::VisibilityFlags::Type  visibility)
+    {
+        // EDF for the light.
+        asr::ParamArray edf_params;
+        edf_params.insert("radiance", color_name);
+        edf_params.insert("radiance_multiplier", intensity);
+
+        const std::string edf_name = make_unique_name(assembly.edfs(), light_name + "_edf");
+        assembly.edfs().insert(
+            asr::DiffuseEDFFactory().create(edf_name.c_str(), edf_params));
+
+        // Material for the light.
+        asr::ParamArray material_params;
+        material_params.insert("edf", edf_name);
+
+        const std::string material_name =
+            make_unique_name(assembly.materials(), light_name + "_mat");
+
+        assembly.materials().insert(
+            asr::GenericMaterialFactory().create(material_name.c_str(), material_params));
+
+        // Light object instance.
+        const std::string instance_name =
+            make_unique_name(assembly.object_instances(), light_name + "_inst");
+        
+        // Always remove DiffuseRay.
+        visibility ^= asr::VisibilityFlags::DiffuseRay;
```

Comment:
I'm not sure why we would want to do that. Can you explain the rationale?

Reply:
I saw that in analytical lights test scenes and I thought it's something that applies to all light objects. Now I know it doesn't. Removing this.